glacierscript.py

```diff
@@ -493,20 +493,35 @@ def write_and_verify_qr_code(name, filename, data):
     Write a QR code and then read it back to try and detect any tricksy malware tampering with it.
 
     name: <string> short description of the data
-    filename: <string> filename for storing the QR code
+    filename: <string> filename for storing the QR code. note that ".png" is added in function to make incrementing easier
     data: <string> the data to be encoded
     """
 
-    subprocess.call("qrencode -o {0} {1}".format(filename, data), shell=True)
+    # note: it would probably be better to write qr codes to RAM disk or secure volume. while shouldn't contain keys, may increase security by decreasing chance of unencrypted transaction history being discovered
+    QR_SUBDIR = "qrcodes"
+    QR_SUFFIX = ".png"
+    script_root = os.path.dirname(os.path.abspath(__file__))
+    QR_DIRPATH = script_root + "/" + QR_SUBDIR
+    if not os.path.isdir(QR_DIRPATH):
+        os.mkdir(QR_DIRPATH)
+    QR_PATH = QR_DIRPATH + "/" + filename + QR_SUFFIX
+    if os.path.exists(QR_PATH):
+        #print "QR exists at: {0}".format(QR_PATH)
+        i = 2
+        while os.path.exists(QR_DIRPATH + "/" + filename + str(i) + QR_SUFFIX):
```

Comment:
Ugly code duplication here -- construction of QR_PATH appears three separate times. That must be in one place (principle of DRY: Don't Repeat Yourself).